src/Microsoft.DotNet.ProjectJsonMigration/Rules/MigrateScriptsRule.cs

```diff
@@ -0,0 +1,229 @@
+                }
+            }
+
+            return command;
+        }
+
+        private bool IsPathRootedForAnyOS(string path)
+        {
+            return path.StartsWith("/") || path.Substring(1).StartsWith(":\\");
+        }
+
+        private ProjectTargetElement CreateTarget(ProjectRootElement csproj, string scriptSetName)
+        {
+            var targetName = $"{scriptSetName[0].ToString().ToUpper()}{string.Concat(scriptSetName.Skip(1))}Script";
+            var targetHookInfo = ScriptSetToMSBuildHookTargetMap[scriptSetName];
+
+            var target = csproj.AddTarget(targetName);
+            if (targetHookInfo.IsRunBefore)
+            {
+                target.BeforeTargets = targetHookInfo.TargetName;
+            }
+            else
+            {
+                target.AfterTargets = targetHookInfo.TargetName;
+            }
+
+            return target;
+        }
+
+        private void AddExec(ProjectTargetElement target, string command)
+        {
+            var task = target.AddTask("Exec");
+            task.SetParameter("Command", command);
+        }
+
+        // ProjectJson Script Set Name to 
+        private static Dictionary<string, TargetHookInfo> ScriptSetToMSBuildHookTargetMap => new Dictionary<string, TargetHookInfo>()
+        {
+            { "precompile",  new TargetHookInfo(true, "Build") },
+            { "postcompile", new TargetHookInfo(false, "Build") },
+            { "prepublish",  new TargetHookInfo(true, "Publish") },
+            { "postpublish", new TargetHookInfo(false, "Publish") }
+        };
+
+        private static Dictionary<string, string> ScriptVariableToMSBuildMap => new Dictionary<string, string>()
+        {
+            { "compile:TargetFramework", null },  // TODO: Need Short framework name in CSProj
+            { "compile:ResponseFile", null },     // Not migrated
+            { "compile:CompilerExitCode", null }, // Not migrated
+            { "compile:RuntimeOutputDir", null }, // Not migrated
+            { "compile:RuntimeIdentifier", null },// TODO: Need Rid in CSProj
+            
+            { "publish:TargetFramework", null },  // TODO: Need Short framework name in CSProj
+            { "publish:Runtime", null },          // TODO: Need Rid in CSProj
+
+            { "compile:FullTargetFramework", "$(TargetFrameworkIdentifier)=$(TargetFrameworkVersion)" },
+            { "compile:Configuration", "$(Configuration)" },
+            { "compile:OutputFile", "$(TargetPath)" },
+            { "compile:OutputDir", "$(TargetDir)" },
+
+            { "publish:ProjectPath", "$(MSBuildThisFileDirectory)" },
+            { "publish:Configuration", "$(Configuration)" },
+            { "publish:OutputPath", "$(TargetDir)" },
+            { "publish:FullTargetFramework", "$(TargetFrameworkIdentifier)=$(TargetFrameworkVersion)" },
+
+            { "project:Directory", "$(MSBuildProjectDirectory)" },
+            { "project:Name", "$(MSBuildThisFileName)" },
```

Comment:
In the context of migration they should both be the same. I'm okay with using AssemblyName though, seems less fragile.